cmd/importuser.go

```diff
@@ -0,0 +1,198 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/nats-io/jwt"
+	"github.com/nats-io/nkeys"
+	"github.com/nats-io/nsc/cmd/store"
+	"github.com/spf13/cobra"
+)
+
+func createImportUserCmd() *cobra.Command {
+	var params ImportUser
+	cmd := &cobra.Command{
+		Use:          "user --file <user-jwt/user-creds>",
+		Short:        "Imports an account from a jwt or account and nkey from a creds file",
+		Example:      `nsc import account --jwt <account-jwt>`,
+		Args:         MaxArgs(0),
+		SilenceUsage: false,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := RunMaybeStorelessAction(cmd, args, &params); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	cmd.Flags().BoolVarP(&params.skip, "skip", "", false, "skip validation issues")
+	cmd.Flags().BoolVarP(&params.overwrite, "overwrite", "", false, "overwrite existing user")
+	cmd.Flags().StringVarP(&params.file, "file", "f", "", "user jwt or creds to import")
+	cmd.MarkFlagRequired("file")
+	return cmd
+}
+
+func init() {
+	importCmd.AddCommand(createImportUserCmd())
+}
+
+type ImportUser fileImport
+
+func (p *ImportUser) SetDefaults(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *ImportUser) PreInteractive(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *ImportUser) Load(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *ImportUser) PostInteractive(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *ImportUser) Validate(ctx ActionCtx) error {
+	fi, err := os.Stat(p.file)
+	if err != nil {
+		return err
+	}
+	if fi.IsDir() {
+		return fmt.Errorf("%#q is a directory", p.file)
+	}
+	if !(strings.HasSuffix(p.file, ".jwt") || strings.HasSuffix(p.file, ".creds")) {
+		return fmt.Errorf("expected .jwt or .creds file")
+	}
+	return nil
+}
+
+// returns true when blocking and not skipped
+func validateAndReport(claim jwt.Claims, force bool, r *store.Report) bool {
+	vr := jwt.ValidationResults{}
+	claim.Validate(&vr)
+	for _, i := range vr.Issues {
+		if i.Blocking {
+			r.AddError("Validation resulted in: %s", i.Description)
+		} else {
+			r.AddWarning("Validation resulted in: %s", i.Description)
+		}
+	}
+	if vr.IsBlocking(true) && !force {
+		r.AddError("Validation is blocking, skip with --skip")
+		return true
+	}
+	return false
+}
+
+func (p *ImportUser) Run(ctx ActionCtx) (store.Status, error) {
+	r := store.NewDetailedReport(true)
+	content, err := ioutil.ReadFile(p.file)
```

Comment:
load should have already loaded the file...

Reply:
moved that to Load